### DISCUSSION TITLE
Padding indicator in data

### DISCUSSION THREAD

**Discussion by Vlieking** [awslabs/gluonts#3048] [Q&A] [answered]
Hello, sorry if this has been asked before; I could not find any discussion about it nor does the documentation make it clear what to do. 

I have a dataset consisting of many different time series, all with a length between 10 and 60. To be able to use these different time series in a single model, I want to add padding to the shorter ones. I can manually add zeros or NaNs to the dataset, but I'm unsure whether the model will recognize this as padding, and I do not want the model to start predicting zeros as a result of this. 

I could only find documentation about adding padding to data using the InstanceSplitter transformation, however, I do not wish to split my data. Is manually adding an "is_padding" field a potential solution? 

Thanks!

**Comment by lostella** [comment, marked as answer]
@Vlieking what are you trying to do with the data?

If you need to train an estimator, then there is no need to pad the data to make it of even length, since estimators can be trained on series of different length in general.

**Reply by Vlieking** [reply]
Ah, I was under the impression that prediction_length and context_length need to be defined to create a working predictor. Can those fields be left undefined in that case? Or perhaps only the context length

**Reply by lostella** [reply]
I think that's not important for the data: sure, most (all?) models have some "context length" that configures how far back in the past the model sees to make predictions, but they usually (always?) take care of padding/cutting the data themselves, both at training time and prediction time
